github_handler.py

```
@@ -0,0 +1,358 @@
+                
+                try:
+                    decoded_content = base64.b64decode(content).decode('utf-8')
+                    
+                    max_length = _get_readme_limit(REPO_README_MAX_LENGTH)
+                    
+                    if len(decoded_content) > max_length:
+                        logger.info(f"Truncating README from {len(decoded_content)} to {max_length} characters")
+                        decoded_content = decoded_content[:max_length] + "\n\n... (README truncated for brevity)"
+                    
+                    logger.info(f"Successfully fetched README for {owner}/{repo} ({len(decoded_content)} chars)")
+                    return decoded_content
```

Ruff (0.14.3)

Consider moving this statement to an `else` block

(TRY300)